### PR TITLE
Lucile - add work confirmation improvements to people report

### PR DESCRIPTION
**Pull request by LucileTech**
###Description
9. (PRIORITY MEDIUM) Jae: Add work confirmation improvements to People Report (WIP Lucile)
Admin or Owner login → Reports → Reports → People → Choose
Remove “hours logged this week” section: If a user is INACTIVE (has an end date), then the “Hours Logged This Week” square should not show. Center the remaining ones.
Please also add the User’s Title under their name.
Please also add a “Rehireable” section under their name that can be clicked as Yes or No

11. (PRIORITY LOW): In smaller view widths (<=1200px), the tops of the report blocks are cut off. (WIP Lucile)

###Related PR
To test this frontend PR you need to checkout the  #362 backend PR.
https://github.com/OneCommunityGlobal/HGNRest/pull/362

###Mainly changes explained:
1 - Was added a new set State in setRehireable in PeapleReport

###Before (sorry for my slow connexion)

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/111132148/e92b0a4c-9f38-4129-963b-645cdada4b9d

###After (sorry for my slow connexion)

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/111132148/ae80a0ca-568f-454f-93f5-adfaa0bd0eef

###How to test:
Git checkout Lucile_add_work_confirmation_improvements_to_people_report in the frontend
do npm install and npm run start:local to run this PR locally

Git checkout Lucile_add_work_confirmation_improvements_to_people_report in the backend
do npm install and npm run build and npm start

Admin or Owner login → Reports → Reports → People → Choose someone with an end date 
You should not see "hours logged this week" section
You should see the User’s Title under their name
You should be able to checkout and checkout the rehireable check box on the right.
The chosen check should remain even if the page is refreshed.
The tops of the report blocks should not be cut off in smaller views

Admin or Owner login → Reports → Reports → People → Choose someone without an end date 
You should see "hours logged this week" section
You should see the User’s Title under their name
You should not see the rehireable checkbox
The tops of the report blocks should not be cut off in smaller views

### CODE REVIEW

File: src/components/Reports/PeopleReport/PeopleReport.jsx [hunk 1]
```diff
@@ -234,66 +225,65 @@ class PeopleReport extends Component {
...
       isActive,
+      isRehireable,
       fromDate,
       toDate,
       timeEntries,
     } = this.state;
     const { firstName, lastName, weeklycommittedHours, totalTangibleHrs } = userProfile;
-
-    var totalTangibleHrsRound = 0
+    console.log('toto', userProfile.isRehireable);
```
Comment (20chen-7) [inline src/components/Reports/PeopleReport/PeopleReport.jsx]: The usage of 'console.log' to track and debug the value of 'userProfile.isRehireable' is indeed helpful for monitoring and troubleshooting purposes. And the logging statement is executed at the appropriate time after assigned values,  ensuring accurate tracking of the variable.
It's also worth mentioning that you're aware of the need to remove or disable these logging statements in the future to prevent unnecessary console output and any potential impact on performance. This demonstrates good practice in terms of code cleanliness and optimization.

**Review by 20chen-7 — COMMENTED** [review]
Consistency in styling, such as using single quotes for variables and double quotes for className, can indeed improve readability and organization. It's a good practice to maintain a consistent coding style throughout your codebase.

Regarding the addition of a checkbox and tracking the isRehireable process with console logging, it's beneficial for monitoring and debugging purposes. However, if the log values only show True, it may be helpful to consider including some false test cases as well. Including test cases for both true and false scenarios allows for better validation of the code's functionality and can help identify any potential issues or edge cases that may need to be addressed.

**Review by 20chen-7 — COMMENTED** [review]
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/102696148/e2c7a406-cdc4-4b76-8eb3-774b013cd2fd
1. Inactive Users and Removed Hours Logged This Week Section: During testing, I noticed that there are some inactive users like the ID "12341234", when I clicked the user, the "Hours Logged This Week" section is still there. 
2. Website Refresh: The removal of the "Hours Logged This Week" section exhibited slower loading times compared to other sections of the website.

**Review by 20chen-7 — DISMISSED** [review]
Looks much better, the function is intended.
Screenshot 2023-05-22 at 1 40 49 PM

**Comment by LucileTech** [comment]
> Looks much better, the function is intended. Screenshot 2023-05-22 at 1 40 49 PM

Thank you for your detailed comments on the PR @20chen-7

**Review by NidaZaki — DISMISSED** [review]
The function works as intended.
For a user without the end date:
PR#836
For a user with the end date : (rehireable checkbox)
PR#836-1
For the same user, on setting the rehireable checkbox to ON and refreshing the page, the end date for that user changes and the "hours logged this week" section appears:
PR#836-2
If the rehireable checkbox is OFF and the page is refreshed, the checkbox remains as it is. (refer the image below)
PR#836-before refresh

**Review by stevenpaalz — CHANGES_REQUESTED** [review]
Please see my note on [PR #362.](https://github.com/OneCommunityGlobal/HGNRest/pull/362#pullrequestreview-1440604785) I am not able to replicate expected behavior.

**Review by nathanah — APPROVED** [review]
Changes appear to work as intended, but, in smaller views (width<=1200px), the tops of the report blocks are cut off.
image

**Review by OAkinbode — DISMISSED** [review]
The features work as intended. The only thing I see is that the "total hours logged this week" component loads up for a short while and then disappears. As below. Not sure it is a deal though.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/104801652/28354cc3-3fe6-4075-bae9-ff185d28c5cd

**Comment by stevenpaalz** [comment]
> Please see my note on [PR #362.](https://github.com/OneCommunityGlobal/HGNRest/pull/362#pullrequestreview-1440604785) I am not able to replicate expected behavior.

This has been resolved

**Review by stevenpaalz — APPROVED** [review]
With latest backend commit this is now working as intended.
Screenshot 2023-05-24 at 9 01 40 AM

**Comment by LucileTech** [comment]
> Changes appear to work as intended, but, in smaller views (width<=1200px), the tops of the report blocks are cut off. image

Thank you [nathanah](https://github.com/nathanah), I fixed it

**Comment by eduhorta** [comment]
Hi @LucileTech!

It appears to be functioning correctly.
The weekly hours logged display briefly when accessing an user with an end date.
I have a question:
In the video below, I checked the rehireable box for a user with an end date and refreshed the page. However, the box disappeared and the weekly hours logged value reappeared.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105796041/41f8ae8c-800a-4400-ae21-c80ca15a01df

I could not reproduce this:
image

File: src/components/Reports/PeopleReport/PeopleReport.jsx [hunk 2]
```diff
@@ -460,24 +449,38 @@ class PeopleReport extends Component {
...
+        <p>{userProfile.role}</p>
+
+        {userProfile.endDate ? (
+          <p>
+            <Checkbox
+              value={this.state.isRehireable}
+              onChange={() => this.setRehireable(!this.state.isRehireable)}
+              label="Rehireable"
+            ></Checkbox>
+          </p>
```
Comment (nathanah) [inline src/components/Reports/PeopleReport/PeopleReport.jsx:461]: Warning: validateDOMNesting(...): &lt;div&gt; cannot appear as a descendant of &lt;p&gt;.
```suggestion
          <div>
            <Checkbox
              value={this.state.isRehireable}
              onChange={() => this.setRehireable(!this.state.isRehireable)}
              label="Rehireable"
            ></Checkbox>
          </div>
```

File: src/components/Reports/sharedComponents/ReportPage/ReportPage.css [hunk 3]
```diff
@@ -20,4 +19,4 @@
   padding: 24px 24px 0 24px;
   background-color: #f1ebf7;
   border-top-left-radius: 25px;
-}
\ No newline at end of file
+}
```
Comment (nathanah) [inline src/components/Reports/sharedComponents/ReportPage/ReportPage.css:22]:
```suggestion
}

@media(max-width:850px){
  .report-page-wrapper {
    display: flex;
    flex-direction: column;
    padding: 40px;
    align-items: center;
  }

  .report-page-profile {
    padding: 24px;
    border-radius: 25px;
  }

  .report-page-content {
    padding: 24px;
  }
}
```

Widths below 850px have the profile component fall off the screen.
image

**Review by nathanah — CHANGES_REQUESTED** [review]

**Review by yoyoweiii — COMMENTED** [review]
Works as intended. 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/60588210/289d4c65-a6b7-40cd-8f8c-f4d0be9ca6cf

**Comment by AnishPandita** [comment]
Hi @LucileTech I tested your with the specified backend branch and its working as intended. Great work.
Screenshot 2023-05-24 at 7 51 51 PM
Screenshot 2023-05-24 at 7 52 17 PM

**Comment by LucileTech** [comment]
> Hi @LucileTech!
> 
> It appears to be functioning correctly. The weekly hours logged display briefly when accessing an user with an end date. I have a question: In the video below, I checked the rehireable box for a user with an end date and refreshed the page. However, the box disappeared and the weekly hours logged value reappeared.
> 
>  2023-05-24.13-28-09.mp4 
> I could not reproduce this: image

Hi @eduhorta did you pull the backend branch? It happened with someone else and he just didn't have the last commit on the backend branch. 
Please git pull / build / start on the backend branch and try again if you have time, you should have on line 311 of the userProfileController :
if (yearMonthDayDateValidator(req.body.endDate) || new Date(req.body.endDate)) {

**Comment by LucileTech** [comment]
Thank you very much for your suggestions @nathanah, I followed them and change the <p> and added media queries for views under 850px

**Review by lchua2314 — CHANGES_REQUESTED** [review]
Hi @LucileTech I made a review on the backend. Please make the small changes here and the backend and I'll approve both PRs. Good work.

**Comment by LucileTech** [comment]
> Hi @LucileTech I made a review on the backend. Please make the small changes here and the backend and I'll approve both PRs. Good work.

Thank you @lchua2314 I fixed the issues mentionned.

**Review by eduhorta — DISMISSED** [review]
Hi @LucileTech!
I did test the feature with the latest commit on both front/back and everything seems to be working as intended.
Nice job! 👍

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105796041/1b6a7da7-1508-42df-8821-c01692dada0c

**Review by nathanah — APPROVED** [review]

**Review by lchua2314 — DISMISSED** [review]

**Comment by one-community** [comment]
Branch conflicts need to be resolved 
image

**Comment by one-community** [comment]
"Please also add the User’s Title under their name." is referring to their professional title, not their role. Go ahead and leave their role there too though, since you've already added it. Please also add their professional title (the one entered during team member setup and found on their Profile page here: 
image

**Comment by one-community** [comment]
Please also make clicking their picture take a the viewer to their Profile page in case I realize their title or other details need to be edited: 
image

**Comment by LucileTech** [comment]
Hi NidaZaki, thank you for your review. Did you git push on the backend branch? It should not act like that and people who had this issue did not push the backend branch. After pushing it it worked as shown on the description.
Thank you

**Comment by LucileTech** [comment]
> This PR works as intended. But I had a question, I dont know if this is out of scope for this PR, if we check the "Rehireable" box, and refresh the page, the "Hours logged this week" section appears, the "Rehireable" checkbox box disappears, and the end date for that user is "N/A", meaning that user is active. So in the table, can we change the dot color for that user to green
> 
> PR#836
> 
> Please refer the video
> 
>  PR.836.mov

Hi @NidaZaki, thank you for your review. Did you git pull on the backend branch? It should not act like that and people who had this issue did not pull the backend branch after checkout. After pulling it, it should work as shown in the description.
Thank you

**Review by NidaZaki — DISMISSED** [review]

**Comment by one-community** [comment]
Thank you all, moving this to final review.
